tools/dart/dart_roll_helper.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python
+# Copyright 2018 The Dart project authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This script automates the Dart SDK roll steps, including:
+#   - Updating the Dart revision in DEPS
+#   - Updating the Dart dependencies in DEPS
+#   - Syncing dependencies with 'gclient sync'
+#   - Generating GN files for relevant engine configurations
+#   - Building relevant engine configurations
+#   - Running tests in 'example/flutter_gallery' and 'packages/flutter'
+#   - Launching flutter_gallery in release and debug mode
+#   - Running license.sh and updating license files in
+#     'flutter/ci/licenses_golden'
+#   - Generating a commit with relevant Dart SDK commit logs (optional)
+#
+# The following environment variables can be set:
+#   - FLUTTER_HOME: the absolute path to the 'flutter' directory
+#   - ENGINE_HOME: the absolute path to the 'engine/src' directory
+#   - DART_SDK_HOME: the absolute path to the root of a Dart SDK project
+
+import argparse
+import datetime
+import fileinput
+import os
+import shutil
+import subprocess
+import sys
+
+def env_var(var):
+  try:
+    return os.environ[var]
+  except KeyError:
+    return ''
+
+FLUTTER_HOME = env_var('FLUTTER_HOME')
+ENGINE_HOME = env_var('ENGINE_HOME')
+DART_SDK_HOME = env_var('DART_SDK_HOME')
+
+MAX_GCLIENT_RETRIES = 3
+
+ERROR_GCLIENT_SYNC_FAILED    = 1
+ERROR_BUILD_FAILED           = 2
+ERROR_PKG_FLUTTER_FAILED     = 3
+ERROR_FLUTTER_GALLERY_FAILED = 4
+
+def engine_golden_licenses_path():
+  return os.path.join(ENGINE_HOME, 'flutter/ci/licenses_golden/')
```

Comment:
os.path.join(ENGINE_HOME, 'flutter', 'ci', 'licenses_golden') here and below

Reply:
Done.